stl/inc/xstring

```diff
@@ -34,6 +40,90 @@ struct _Char_traits { // properties of a string or stream element
     using off_type   = streamoff;
     using state_type = _Mbstatet;
 
+    // For copy/move, we can uniformly call memcpy/memmove (or their builtin versions) for all element types.
+
+    static _CONSTEXPR20 _Elem* copy(_Out_writes_all_(_Count) _Elem* const _First1,
+        _In_reads_(_Count) const _Elem* const _First2, const size_t _Count) noexcept /* strengthened */ {
+        // copy [_First2, _First2 + _Count) to [_First1, ...)
+#if _HAS_MEMCPY_MEMMOVE_INTRINSICS
+        __builtin_memcpy(_First1, _First2, _Count * sizeof(_Elem));
+#else // ^^^ _HAS_MEMCPY_MEMMOVE_INTRINSICS ^^^ / vvv !_HAS_MEMCPY_MEMMOVE_INTRINSICS vvv
+#ifdef __cpp_lib_is_constant_evaluated
+        if (_STD is_constant_evaluated()) {
+            // pre: [_First1, _First1 + _Count) and [_First2, _First2 + _Count) do not overlap; see LWG-3085
+            for (size_t _Idx = 0; _Idx < _Count; ++_Idx) {
+                _First1[_Idx] = _First2[_Idx];
+            }
+
+            return _First1;
+        }
+#endif // __cpp_lib_is_constant_evaluated
+
+        _CSTD memcpy(_First1, _First2, _Count * sizeof(_Elem));
+#endif // ^^^ !_HAS_MEMCPY_MEMMOVE_INTRINSICS ^^^
+
+        return _First1;
+    }
+
+    _Pre_satisfies_(_Dest_size >= _Count) static _Elem* _Copy_s(_Out_writes_all_(_Dest_size) _Elem* const _First1,

```

Comment:
`_CONSTEXPR20`? It seems peculiar that `_Copy_s` isn't `constexpr` when `copy` is.

Reply:
Will submit a followup PR.